test/e2e/tests/metrics/app-opened.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { strict as assert } from 'assert';
+import { Mockttp } from 'mockttp';
+import {
+  withFixtures,
+  getEventPayloads,
+  unlockWallet,
+  connectToDapp,
+} from '../../helpers';
+import FixtureBuilder from '../../fixture-builder';
+
+/**
+ * Mocks the segment API for the App Opened event that we expect to see when
+ * these tests are run.
+ *
+ * @param mockServer - The mock server instance.
+ * @returns The mocked endpoints
+ */
+async function mockSegment(mockServer: Mockttp) {
+  return [
+    await mockServer
+      .forPost('https://api.segment.io/v1/batch')
+      .withJsonBodyIncluding({
+        batch: [{ type: 'track', event: 'App Opened' }],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+  ];
+}
+
+describe('App Opened metric @no-mmi', function () {
+  it('should send AppOpened metric when app is opened and metrics are enabled', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-fd20',
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        title: this.test?.fullTitle(),
+        testSpecificMock: mockSegment,
+      },
+      async ({ driver, mockedEndpoint: mockedEndpoints }) => {
+        await unlockWallet(driver);
+
+        const events = await getEventPayloads(driver, mockedEndpoints);
+        assert.equal(events.length, 1);
+        assert.equal(events[0].properties.category, 'App');
```

Comment:
Can we also move them to POM helper flow and use it as something like `await homepage.waitForAppOpenedEventSent` ? 
you can find more guideline in 
- [contribution guidelines here](https://docs.google.com/document/d/1Hi64lJ8ZvTXNaoJZeh23_7HjhkRKtlcXrnYZKtQWeWk/edit#heading=h.y7td1yioh1l3)
- [best practices in contributor-doc](https://github.com/MetaMask/contributor-docs/blob/main/docs/e2e/extension-e2e-guidelines.md#best-practices)

Reply:
Danica, are you referring to these lines?

```
const events = await getEventPayloads(driver, mockedEndpoints);
assert.equal(events.length, 1);
```

If so, they’re tied to the `mockSegment` function in this file. Could you share more about your suggestion and the reasoning behind it? Happy to discuss!

Reply:
Danica, was [this reply from you](https://github.com/MetaMask/metamask-extension/pull/28927#discussion_r1876338784) intended as a response in this comment thread?

If so, thanks for the suggestion! Including a method on the home page object to assert that the app opened event was fired seems a bit tricky in this case. Since the app opened event isn't necessarily tied to the home screen (e.g., it could occur on a confirmation screen), and because this assertion is quite specific to this test and the mock segment.io request/response, it might make sense to handle it differently. What do you think?

Reply:
yes sorry my bad Derek, i replied to a different thread ! Yeah makes sense in terms of tying to no specific screen 🙏🏻